aws/rust-runtime/aws-types/src/config.rs

```diff
@@ -60,6 +63,42 @@ impl Builder {
         self
     }
 
+    /// Set the retry_config for the builder
+    ///
+    /// # Examples
+    /// ```rust
+    /// use aws_types::config::Config;
+    /// use smithy_types::retry::RetryConfig;
+    /// let config = Config::builder().retry_config(RetryConfig::new()).build();
+    /// ```
+    pub fn retry_config(mut self, retry_config: impl Into<Option<RetryConfig>>) -> Self {
+        self.set_retry_config(retry_config);
+        self
+    }
+
+    /// Set the retry_config for the builder
+    ///
+    /// # Examples
+    /// ```rust
+    /// use aws_types::config::Config;
+    /// use smithy_types::retry::RetryConfig;
+    ///
+    /// fn retry_config_override() -> Option<RetryConfig> {
+    ///     // ...
+    ///     # None
+    /// }
+    ///
+    /// let mut builder = Config::builder();
+    /// if let Some(retry_config) = retry_config_override() {
+    ///     builder.set_retry_config(retry_config);
+    /// }
```

Comment:
I don't think the if statement around `set_retry_config` is necessary in this example. Maybe a better example would be one where the builder is passed via mutable reference to another function that fills things in on it? Otherwise, I don't think it really drives home the difference from `retry_config`.